types/hooks/appHooks.d.ts

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License.
+
+import { HookContext, HookContextInit } from './HookContext';
+
+/**
+ * Handler for app start hooks
+ */
+export type AppStartHandler = (context: AppStartContext) => void | Promise<void>;
+
+/**
+ * Context on a function app during app startup.
```

Comment:
There's inconsistency with the doc comments in this file. Sometimes they end with a period, sometimes they don't. We should try to remain consistent.

Reply:
Yeah I go back and forth in my own head on this. Overall I like using a period, but I also feel weird using a period when the comment isn't a full sentence

Regardless, for stylistic choices like this I usually say we enforce it automatically or we don't worry about it. We use eslint/prettier to do that, but unfortunately I'm not finding a rule to enforce a period (or no period) at the end of a doc comment

Reply:
For clauses like `Context on a function app during app startup`, we don't need to worry about it. There are some full sentences with a period and some without. I think those should be the only ones we try to clean up. I don't think ESLint or Prettier will cover grammatical/style issues like this, however.

Reply:
To clarify, I agree a period is technically correct, but it's not my default style choice (see my use of periods in this thread so far - I swear I didn't do that on purpose lol). I'm probably the source of the sentences without periods and Hossam probably did the other ones. Periods aren't required for understanding the content and imply a certain level of formality/seriousness/emotion that I prefer to avoid. I'm [not alone](https://lifehacker.com/dont-use-periods-in-texts-1843744818) in this habit, either

So if we find an automated eslint/prettier rule I'm willing to go against my style preference, but otherwise I don't want to change my habits for something I don't necessarily like anyways

Reply:
Merging as-is, but we can continue to discuss outside this PR if you want